legion/tests/test_ltemplate.py

```diff
@@ -0,0 +1,117 @@
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+"""
+Legion Templating System
+"""
+
+import unittest2
+import tempfile
+import os
+import _thread
+import time
+import asyncio
+from legion.templating import TemplateSystem
+import yaml
+
+
+class TestTemplateSystem(unittest2.TestCase):
+    """
+    Unit tests for template system
+    """
+
+    TEMPLATE_LINE = '<file>%s</file>'
+
+    LOAD_MODULE_LINE = "{{ load_module('legion.io.render_on_file_change', filepath='%s', var_name='data') }}"
+    TEMPLATE_PRINT_VARIABLE = "{{ data }}"
+
+    LOAD_YAML_MODULE_LINE = "{{ load_module('legion.io.render_on_file_change', filepath='%s', " \
+                            "var_name='data', is_yaml_file=True) }}"
+    TEMPLATE_PRINT_YAML_VARIABLE = '{% if data is defined %}{{ data["item"] }}{% endif %}'
+
+    SLEEP_INTERVAL_IN_SEC = 3
+
+    def setUp(self):
+        """
+        Set up test data
+        """
+        self.random_data = str(time.time())  # create random string
+        self.output_file, self.output_path = tempfile.mkstemp()  # temp output file for file watch
+        self.input_file, self.input_path = tempfile.mkstemp()  # temp input file for file watch
+        self.template_file, self.template_path = tempfile.mkstemp()  # temp template file for file watch
+
+        with os.fdopen(self.template_file, 'w') as template:  # create template file for file watch
+            template.write(self.LOAD_MODULE_LINE % self.input_path)
+            template.write(self.TEMPLATE_LINE % self.TEMPLATE_PRINT_VARIABLE)
+
+        self.output_yaml_file, self.output_yaml_path = tempfile.mkstemp()  # temp output file for yaml file watch
+        self.input_yaml_file, self.input_yaml_path = tempfile.mkstemp()  # temp input file for yaml file watch
+        self.template_yaml_file, self.template_yaml_path = tempfile.mkstemp()  # temp template file for yaml file watch
+
+        with os.fdopen(self.template_yaml_file, 'w') as template:  # create template file for yaml file watch
+            template.write(self.LOAD_YAML_MODULE_LINE % self.input_yaml_path)
+            template.write(self.TEMPLATE_LINE % self.TEMPLATE_PRINT_YAML_VARIABLE)
+
+    def _template_render_loop(self):
+        """
+        Start a thread with a render loop for file watch test
+        """
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
+        ts = TemplateSystem(self.template_path, self.output_path)
+        ts.render_loop()
+
+    def test_file_watch(self):
+        """
+        Test a file watch module
+        """
+        th = _thread.start_new_thread(self._template_render_loop, ())  # start a thread
```

Comment:
Why you dont stop thread? I whink would be better if you will use context manager for thread procedures (start & end)